### PR TITLE
Feature/ioda sprint atms hdf5

### LINKED ISSUE #841: update the ATMS hdf5 converter to the IODA conventions
The ATMS hdf5 IODA converter needs updates to bring it in line with the latest IODA conventions.

This will need changes to both the `ioda-converter` and `ioda` repositories themselves.

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

This update the python HDF5 converter to the latest IODA conventions

### Issue(s) addressed

- fixes #841 

## Acceptance Criteria (Definition of Done)

successful pass of the ctest `test_iodaconv_atms_nc4`

## Dependencies

- [x] waiting on JCSDA-internal/ioda#633
- [x] waiting on jcsda-internal/ioda/pull/648

### CODE REVIEW

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (gthompsnJCSDA) [inline src/hdf5/atms_netcdf_hdf5_2ioda.py:296]: I previously commented on this but this units does not align with a string; only when dateTime is filled with integers

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (gthompsnJCSDA) [inline src/hdf5/atms_netcdf_hdf5_2ioda.py:265]: There are a very large number of lines with hard-coded group names of ```MetaData```, ```ObsValue```, ```ObsError```, and ```PreQc```.  If you review my python code, we can take simple care of making a set of 4 variables to query what group names are in IODA standard to make future hard-coded names less necessary.  Please review the 4 lines of code starting at [amdar_bufr2ioda.pyL84](https://github.com/JCSDA-internal/ioda-converters/blob/7b46bc7f4401e539639a6e1f8142ef706cc1d8b7/src/conventional/amdar_bufr2ioda.py#L84).  If you capture the group names and re-use them later, then literally no code changes are needed if some future IODA version changes group names.

**Review by gthompsnJCSDA — CHANGES_REQUESTED** [review]

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:32]: same as above -- I'm not keen to import another python library just to get missing values, was hoping to replace this when can inherit these values from IODA.

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:46]: great thanks!

Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:46]: but these "string" and "float" are being passed directory to the IODA writer what should the "int64" label be?

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: yes will change in a subsequent pull request but just used what was available

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:196]: yes should be global attribute but would like to fix the core of what's wrong and defer this to the next batch of changes

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:296]: yup

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:261]: sure would like to defer

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:265]: yes would like to defer to another issue and pull request

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: this is done

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (CoryMartin-NOAA) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:32]: There is `float_missing_value = iconv.get_default_fill_val(np.float)`

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (CoryMartin-NOAA) [inline src/hdf5/atms_netcdf_hdf5_2ioda.py:259]: Is this not more confusing than doing:
```
if write_string:
else:
```
and flip the if and else?

**Comment by CoryMartin-NOAA** [comment]
@BenjaminRuston why is there an input file deleted in this PR? Is that the cause of the issue?

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: done

**Comment by BenjaminRuston** [comment]
> @BenjaminRuston why is there an input file deleted in this PR? Is that the cause of the issue?

note I removed an original way too large input file, Steve H. made a sampled subset of the data which is what is used in the ctest

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: great using it thanks

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: done following Cory showing how to use it from IODA

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: got to it and all done

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: sure flipped and just created string time once

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (CoryMartin-NOAA) [inline src/hdf5/atms_netcdf_hdf5_2ioda.py:262]: Does this run? Where is `cdtg` defined?

Comment (CoryMartin-NOAA) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py:262]: `observation_time_string` instead?

File: src/hdf5/atms_netcdf_hdf5_2ioda.py
Comment (srherbener) [inline src/hdf5/atms_netcdf_hdf5_2ioda.py:90]: Is this an error condition that warrants stopping execution? If so it should raise an Exception or exit and it would be helpful to indicate in the message that this is an error. If not, the message should indicate that this is a warning. Thanks!

Comment (BenjaminRuston) [reply src/hdf5/atms_netcdf_hdf5_2ioda.py]: Steve added  `ERROR` fixed the check as it would have never hit

and added an exit

**Review by rhoneyager — APPROVED** [review]
LGTM

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by srherbener — APPROVED** [review]
Thanks for addressing my comment.

**Comment by srherbener** [comment]
When I test on my Mac with all of the feature/sprint-ioda-converters branches for ioda and ioda-data I'm seeing test failures for this PR.
```
593/706 Testing: test_iodaconv_atms_nc4
593/706 Test: test_iodaconv_atms_nc4
Command: "/usr/local/bin/bash" "/Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/build/bin/iod
aconv_comp.sh" "netcdf" "/usr/local/Frameworks/Python.framework/Versions/3.9/bin/python3.9 /Users/st
eveherbener/projects/CONVERTER_V2/ioda-bundle/iodaconv/src/hdf5/atms_netcdf_hdf5_2ioda.py
                          -d 2021120600
                          -i testinput/SNDR_SNPP_ATMS_SAMPLE.nc4
                          -o testrun/2021120600_atms_sdr.nc4" "2021120600_atms_sdr.nc4" "0.0"
Directory: /Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/build/iodaconv/test
"test_iodaconv_atms_nc4" start time: Mar 11 13:53 MST
Output:
----------------------------------------------------------
DIFFER : VARIABLE : dateTime : ATTRIBUTE : _FillValue : VALUES : 253370790000000000 <> -4883630631933722624
DIFFER : VARIABLE : dateTime : POSITION : [0] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [1] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [2] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [3] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [4] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [5] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [6] : VALUES : 1638777600 <> 1638774000
DIFFER : VARIABLE : dateTime : POSITION : [7] : VALUES : 1638777600 <> 1638774000
...
Variable Group     Count    Sum AbsSum  Min  Max Range Mean StdDev
dateTime /MetaData   100 360000 360000 3600 3600     0 3600      0
<end of output>
Test time =   0.82 sec
----------------------------------------------------------
Test Failed.
"test_iodaconv_atms_nc4" end time: Mar 11 13:53 MST
"test_iodaconv_atms_nc4" time elapsed: 00:00:00
----------------------------------------------------------


615/706 Testing: iodaconv_validate_testoutput_2021120600_atms_sdr.nc4
615/706 Test: iodaconv_validate_testoutput_2021120600_atms_sdr.nc4
Command: "/Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/build/bin/ioda-validate.x" "/Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/ioda/share/ioda/yaml/validation/ObsSpace.yaml" "/Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/iodaconv/test/testoutput/2021120600_atms_sdr.nc4"
Directory: /Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/build/iodaconv/test
"iodaconv_validate_testoutput_2021120600_atms_sdr.nc4" start time: Mar 11 13:53 MST
Output:
----------------------------------------------------------
Reading YAML from /Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/ioda/share/ioda/yaml/validation/ObsSpace.yaml
ESC[0mProcessing data file: /Users/steveherbener/projects/CONVERTER_V2/ioda-bundle/iodaconv/test/testoutput/2021120600_atms_sdr.nc4
ESC[0m Verifying that all required groups exist
ESC[0m Verifying group ObsValue
ESC[0m Verifying group ObsError
ESC[0m Verifying group MetaData
ESC[0m Verifying group PreQC
ESC[0m Verifying group /
ESC[0mESC[1mESC[34m  Warning: Attribute 'platformCommonName' is present but is not listed as a required or optional attribute in the spec.
ESC[0mESC[1mESC[34m  Warning: Attribute 'platformLongDescription' is present but is not listed as a required or optional attribute in the spec.
ESC[0mESC[1mESC[34m  Warning: Attribute 'platformCommonName' does not have a YAML spec.
ESC[0mESC[1mESC[34m  Warning: Attribute 'platformLongDescription' does not have a YAML spec.
ESC[0m Verifying dimension names
ESC[0m Verifying variable information
ESC[0m  Variable ObsValue/brightnessTemperature
ESC[0m  Variable ObsError/brightnessTemperature
ESC[0m  Variable MetaData/sensorZenithAngle
ESC[0m  Variable MetaData/sensorScanPosition
ESC[0m  Variable MetaData/longitude
ESC[0m  Variable MetaData/latitude
ESC[0m  Variable MetaData/sensorViewAngle
ESC[0m  Variable MetaData/solarZenithAngle
ESC[0m  Variable MetaData/sensorChannelNumber
ESC[0m  Variable MetaData/sensorAzimuthAngle
ESC[0m  Variable MetaData/fieldOfViewNumber
ESC[0m  Variable MetaData/dateTime
ESC[0m  Variable MetaData/solarAzimuthAngle
ESC[0m  Variable PreQC/brightnessTemperature
ESC[0mESC[4mFinal results:ESC[0mESC[31m
  # errors:      0ESC[0mESC[34m
  # warnings:    4
ESC[0m
CTEST_FULL_OUTPUT
<end of output>
Test time =   0.16 sec
----------------------------------------------------------
Test Failed.
"iodaconv_validate_testoutput_2021120600_atms_sdr.nc4" end time: Mar 11 13:53 MST
"iodaconv_validate_testoutput_2021120600_atms_sdr.nc4" time elapsed: 00:00:00
----------------------------------------------------------
```

I know it's a pain with the CI testing not picking up the desired branches, but I'm thinking that these two tests should pass (in a proper environment) before merging.

**Comment by srherbener** [comment]
The test failures I'm seeing on my Mac are due to a defect in the ioda python API where local times are getting confused with UTC (#864) I'm working on a fix for that and should have a PR soon.

**Pull request merged**
